### PR TITLE
feat(redshift): IAM roles can be attached to a cluster, post creation

### LINKED ISSUE #22632: aws-redshift-alpha: create method that adds IAM role to cluster
### Describe the feature

Create a method that will allow users to add an IAM role to an already created cluster. This would allow a user to do the following:
`myCluster.add_role(myRole)`

### Use Case

I'm always frustrated when I can't add a role to an already created cluster, and instead need to add it to the cluster on creation. 

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.25.0a0

### Environment details (OS name and version, etc.)

MacOS Monterey 12.5

### PR DESCRIPTION
**Pull request by Rizxcviii**
Created an `addIamRole` method that will allow attaching an IAM role to a cluster, post its creation.
closes #22632 

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Construct Runtime Dependencies:

* [ ] This PR adds new construct runtime dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-construct-runtime-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-redshift/lib/cluster.ts [hunk 1]
```diff
@@ -662,4 +663,54 @@ export class Cluster extends ClusterBase {
...
+   * @param role the role to add
+   */
+  public addIamRole(role: iam.IRole): void {
+    const clusterRoleList = this.cluster.iamRoles ?? [];
+
+    if (clusterRoleList.includes(role.roleArn)) {
+      throw new Error('Role is already attached to the cluster');
+    }
+    if (clusterRoleList.length >= 10) {
+      throw new Error('Maximum number of IAM roles for a cluster is 10');
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-redshift/lib/cluster.ts:679]: According to https://docs.aws.amazon.com/redshift/latest/mgmt/amazon-redshift-limits.html, the limit is 50 in most regions.

I'd rather not do region-specific weirdness, and instead just let it default to 50.

File: packages/@aws-cdk/aws-redshift/lib/cluster.ts [hunk 2]
```diff
@@ -662,4 +663,54 @@ export class Cluster extends ClusterBase {
...
+  /**
+   * Adds a role to the cluster
+   *
+   * @param role the role to add
+   */
+  public addIamRole(role: iam.IRole): void {
+    const clusterRoleList = this.cluster.iamRoles ?? [];
+
+    if (clusterRoleList.includes(role.roleArn)) {
+      throw new Error('Role is already attached to the cluster');
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-redshift/lib/cluster.ts:676]:
```suggestion
      throw new Error(`Role '${role.roleArn}' is already attached to the cluster`);
```

**Review by comcalvi — CHANGES_REQUESTED** [review]

**Review by comcalvi — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-redshift/lib/cluster.ts (hunk 2 above)
Comment (Rizxcviii) [reply packages/@aws-cdk/aws-redshift/lib/cluster.ts]: Good idea. I don't know why I didn't implement this before. I've just done it

File: packages/@aws-cdk/aws-redshift/lib/cluster.ts [hunk 3]
```diff
@@ -729,8 +737,24 @@ export class Cluster extends ClusterBase {
       policy: AwsCustomResourcePolicy.fromSdkCalls({
         resources: AwsCustomResourcePolicy.ANY_RESOURCE,
       }),
+      installLatestAwsSdk: false,
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-redshift/lib/cluster.ts:740]: good catch!

File: packages/@aws-cdk/aws-redshift/lib/cluster.ts [hunk 4]
```diff
@@ -470,6 +470,13 @@ export class Cluster extends ClusterBase {
    */
   protected parameterGroup?: IClusterParameterGroup;
 
+  /**
+   * The ARNs of the roles that will be attached to the cluster.
+   *
+   * **NOTE** Please do not access this directly, use the `addIamRole` method instead.
+   */
+  private readonly roleArns: string[];
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-redshift/lib/cluster.ts:478]:
```suggestion
  private readonly roles: iam.IRole[];
```

**Review by comcalvi — DISMISSED** [review]
One small comment and this should be ready to merge

File: packages/@aws-cdk/aws-redshift/lib/cluster.ts (hunk 4 above)
Comment (Rizxcviii) [reply packages/@aws-cdk/aws-redshift/lib/cluster.ts]: This has been done

File: packages/@aws-cdk/aws-redshift/lib/cluster.ts (hunk 3 above)
Comment (comcalvi) [reply packages/@aws-cdk/aws-redshift/lib/cluster.ts:740]: It's possible that it never did and that it was committed with `--dry-run`. It's also possible that this used to work when  it was committed and that it no longer works.

**Review by comcalvi — APPROVED** [review]

**Pull request merged**